src/java/org/apache/cassandra/cql3/statements/ModificationStatement.java

```diff
@@ -148,6 +148,14 @@ public ModificationStatement(StatementType type,
                 requiresReadBuilder.add(operation.column);
             }
         }
+        for (ReferenceOperation operation : operations.refs())
+        {
+            if (operation.requiresRead())
+            {
+                conditionColumnsBuilder.add(operation.receiver());
+                requiresReadBuilder.add(operation.receiver());
```

Comment:
this doesn't actually matter as this is used to say if the statement as a whole needs to read... but we don't want that in a txn... ill look to clean this up when trying to avoid the read